### PR TITLE
Adds conditional app composition to Application.js

### PR DESCRIPTION
**Pull request by awichera**

### CODE REVIEW

**Comment by awichera** [comment]
@mpavelka 
I am not sure how am I supposed to use services inside of the component.
My go to method would be to have app availible inside the component. 
But..

I am not sure what is the proper way to pass a param through router to a component.
So I added one to Application.js object.
```jsx
<route.component {...props} {...route.props} />
```

So then you can Add props like 
```jsx
        app.Router.addRoute({ 
            path: '/auth', 
            exact: true, 
            name: 'Authentication', 
            component: AuthContainer,
            props: {"foo":"bar"},            
        });
```

File: src/containers/Application.js [hunk 1]
```diff
@@ -34,16 +35,19 @@ class Application extends Component {
...
 			const module = new props.modules[i](this);
 			this.Modules.push(module);
 		}
 
-		// Create store
+		// Create store		
 		this.Store = Object.keys(this.ReduxService.Reducers).length > 0
 					? createStore(combineReducers(this.ReduxService.Reducers))
-					: createStore((state) => state)
+					: createStore((state) => state)
```
Comment (mpavelka) [inline src/containers/Application.js]: Pls removoe those `		` spaces at the end of lines. You might want to configure your editor or something

File: src/containers/Application.js [hunk 2]
```diff
@@ -74,46 +78,73 @@ class Application extends Component {
 	}
 
 	render() {
+		//const authn = useSelector(state => state.authn);
+		console.log("this.Store",this.Store)
```
Comment (mpavelka) [inline src/containers/Application.js]: Please don't keep these console logs here.

File: src/containers/Application.js [hunk 3]
```diff
@@ -74,46 +78,73 @@ class Application extends Component {
...
-				</AppHeader>
-				<div className="app-body">
-					<AppSidebar fixed display="lg">
-						<AppSidebarNav navConfig={this.Navigation.getItems()} {...this.props} />
-						<AppSidebarFooter />
-						<AppSidebarMinimizer />
-					</AppSidebar>
-					<main className="main">
-						<AppBreadcrumb appRoutes={this.Router.Routes}/>
+			<Provider store={this.Store}>
```
Comment (mpavelka) [inline src/containers/Application.js]: Huuuge `[							]` at the end of the line

File: src/containers/Application.js [hunk 4]
```diff
@@ -74,46 +78,73 @@ class Application extends Component {
...
+														
+													{!route.hasNoHeader ? (
+														<AppHeader fixed>
+															<Header app={this}/>
+														</AppHeader>
+													) : null}
+
+													<div className="app-body">
+
+														{!route.hasNoSidebar ? (
```
Comment (mpavelka) [inline src/containers/Application.js]: should be `hasSidebar` instead.

File: src/containers/Application.js [hunk 5]
```diff
@@ -74,46 +78,73 @@ class Application extends Component {
...
+																	{...this.props} 
+																/>
+																<AppSidebarFooter />
+																<AppSidebarMinimizer />
+															</AppSidebar>
+														) : null}
+
+														<main className="main">
+															
+															{!route.hasNoBreadcrumb ? (
```
Comment (mpavelka) [inline src/containers/Application.js]: `hasBreadcrumb`

File: src/containers/Application.js [hunk 6]
```diff
@@ -74,46 +78,73 @@ class Application extends Component {
...
+																<AppBreadcrumb appRoutes={this.Router.Routes}/>
+															) : null}
+															<route.component {...props} {...route.props} />
+
+														</main>
+														<AppAside fixed>
+														</AppAside>
+													</div>
+
+													{!route.hasNoFooter ? (
```
Comment (mpavelka) [inline src/containers/Application.js]: `hasFooter`

File: src/containers/Application.js [hunk 7]
```diff
@@ -74,46 +78,73 @@ class Application extends Component {
...
+																<AppSidebarMinimizer />
+															</AppSidebar>
+														) : null}
+
+														<main className="main">
+															
+															{!route.hasNoBreadcrumb ? (
+																<AppBreadcrumb appRoutes={this.Router.Routes}/>
+															) : null}
+															<route.component {...props} {...route.props} />
```
Comment (mpavelka) [inline src/containers/Application.js]: If you want to pass the application object, I would go with passing it explicitly:
`{app: this}`

Comment (mpavelka) [reply src/containers/Application.js]: but you can keep the route.props, might get useful (y)

**Pull request merged**